src/server.h

```diff
@@ -960,22 +959,24 @@ typedef struct replBufBlock {
     char buf[];
 } replBufBlock;
 
-typedef struct dbState {
+typedef struct dbDictState {
     list *rehashing;                       /* List of dictionaries in this DB that are currently rehashing. */
+    int resize_cursor;                     /* Cron job uses this cursor to gradually resize dictionaries. */
     unsigned long long key_count;          /* Total number of keys in this DB. */
     unsigned long long *slot_size_index;   /* Binary indexed tree (BIT) that describes cumulative key frequencies up until given slot. */
-} dbState;
+} dbDictState;
 
-#define DICT_TYPE_MAX 2
-#define MAIN_DICT 0
-#define EXPIRE_DICT 1
+typedef enum dbKeyType {
+    DICT_MAIN,
+    DICT_EXPIRES
+} dbKeyType;
```

Comment:
i think the prefix should be DB, not DICT.
how about this?
```suggestion
typedef enum dbDictType {
    DB_DICT_MAIN,
    DB_DICT_EXPIRES
} dbKeyType;
```

Reply:
I'm fine with this terminology. Hope it's the final one. :)

Reply:
@madolson any thoughts. She liked `dbKeyType` as it signifies if the key would be actual data or expire data. Could we discuss here and finalize ? 

I don't have a strong preference here tbh.

Reply:
I agree with the highest level comment, it should be DB and not dict, since it's on the DB and not the dict.

```suggestion
typedef enum dbKeyType {
    DB_MAIN,
    DB_EXPIRES
} dbKeyType;
```

The other point is I don't want to be exposing that there are two underlying dictionaries for the two types of accesses into the main DB. You are either accessing the main keys or volatile keys.